src/components/CalendarLayout/_components/Calendar/Calendar-hooks.tsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useMemo, useState } from 'react'
+import dayjs, { Dayjs } from 'dayjs'
+import { Select } from 'antd'
+import { UseCalendarProps } from './Calendar-types'
+
+export function useCalendar({ appointments, events }: UseCalendarProps) {
+	const [currentDate, setCurrentDate] = useState(dayjs())
+	const currentYear = dayjs().year()
+	const years = useMemo(
+		() => Array.from({ length: 7 }, (_, i) => currentYear - 1 + i),
+		[currentYear],
+	)
+
+	const firstDayOfMonth = currentDate.clone().startOf('month')
+	const lastDayOfMonth = currentDate.clone().endOf('month')
+
+	const filterEvents = useCallback((value: Dayjs) => {
+		return events.filter((event) => dayjs(event.event_date).isSame(value, 'day'))
+	}, [events])
+
+	const filterAppointments = useCallback((value: Dayjs) => {
+		return appointments.filter((appointment) =>
+			dayjs(appointment.planned_date).isSame(value, 'day'),
+		)
+	}, [appointments])
+
+	const updateCurrentDate = useCallback(
+		(operation: 'subtract' | 'add', value: number, unit: 'month' | 'year') => {
+			setCurrentDate((currentDate) => currentDate[operation](value, unit))
+		},
+		[],
+	)
+
+	const monthOptions = useMemo(() => {
+		const options = []
+		const current = currentDate.clone()
+		const months = []
+		for (let i = 0; i < 12; i++) {
+			months.push(current.locale('fr').month(i).format('MMM'))
+		}
+		for (let index = 0; index < 12; index++) {
+			options.push(
+				<Select.Option className="month-item" key={`${index}`}>
+					{months[index]}
+				</Select.Option>,
+			)
+		}
+		return options
+	}, [currentDate])
```

Comment:
Tu peux simplifier comme ça :
```suggestion
        const monthOptions = useMemo(() => {
		const options = []
		
		for (let i = 0; i < 12; i++) {
		         const month = currentDate.locale('fr').month(i).format('MMM')
		         
		         options.push(
				<Select.Option className="month-item" key={month}>
					{month}
				</Select.Option>,
			)
		}
		return options
	}, [currentDate])
```
N'utilise pas l'index comme key, React n'aime pas ça.

Je te conseille de lire la [doc antd](https://ant.design/components/select#usage-upgrade-after-5110) avant d'utiliser les composants. Il est indiqué dans la doc du composant Select que Select.Option est déprécié.